packages/platform-sdk/test/integration-tests/product-projection/product-projection.test.ts

```diff
@@ -0,0 +1,177 @@
+        (attribute) => attribute.name === 'test-number'
+      )[0].value
+    ).toEqual(10.0)
+    expect(
+      productProjectionResponse.body.masterVariant.attributes.filter(
+        (attribute) => attribute.name === 'test-integer'
+      )[0].value
+    ).toEqual(10)
+    expect(
+      productProjectionResponse.body.masterVariant.attributes.filter(
+        (attribute) => attribute.name === 'test-set-text'
+      )[0].value
+    ).toContain('foo')
+    expect(
+      productProjectionResponse.body.masterVariant.attributes.filter(
+        (attribute) => attribute.name === 'test-set-number'
+      )[0].value
+    ).toContain(11.0)
+    expect(
+      productProjectionResponse.body.masterVariant.attributes.filter(
+        (attribute) => attribute.name === 'test-set-integer'
+      )[0].value
+    ).toContain(11)
+    expect(
+      productProjectionResponse.body.masterVariant.attributes.filter(
+        (attribute) => attribute.name === 'test-enum'
+      )[0].value
+    ).toEqual({ key: 'test', label: 'test' })
+
+    await deleteProduct(product)
+    await deleteProductType(productType)
+    await deleteTaxCategory(taxCategory)
+    await deleteCategory(category)
+  })
+
+  it('should query a product by product projection', async () => {
+    const category = await createCategory()
+    const taxCategory = await createTaxCategory()
+    const productType = await createProductType(productTypeDraftForProduct)
+    const productDraft = await createProductDraft(
+      category,
+      taxCategory,
+      productType,
+      false
+    )
+    const product = await createProduct(productDraft)
+    const productProjectionQueryResponse = await apiRoot
+      .productProjections()
+      .get({
+        queryArgs: {
+          where: 'id=' + '"' + product.body.id + '"',
+          staged: true,
+        },
+      })
+      .execute()
+    expect(productProjectionQueryResponse).not.toBe(null)
+    expect(productProjectionQueryResponse.body.results[0].id).toEqual(
+      product.body.id
+    )
+
+    await deleteProduct(product)
+    await deleteProductType(productType)
+    await deleteTaxCategory(taxCategory)
+    await deleteCategory(category)
+  })
+
+  it('should search a product by product projection', async () => {
+    const category = await createCategory()
+    const taxCategory = await createTaxCategory()
+    const productType = await createProductType(productTypeDraftForProduct)
+    const productDraft = await createProductDraft(
+      category,
+      taxCategory,
+      productType,
+      false
+    )
+    const product = await createProduct(productDraft)
+    const productProjectionSearchResponse = await apiRoot
+      .productProjections()
+      .search()
+      .get({
+        queryArgs: {
+          facet: 'categories.id',
+          staged: true,
+        },
+      })
+      .execute()
+    expect(productProjectionSearchResponse).not.toBe(null)
+    expect(productProjectionSearchResponse.body.facets).not.toBe(null)
```

Comment:
This test might be flaky since search indexing is required and it takes a while. Also it does not ensure that the search indexing is activated for the project.

Reply:
I would also put this test in the task that I opened since I don't have a lot of time to fix everything 
I'm sorry :(

Reply:
this is the task opened https://commercetools.atlassian.net/browse/DEVX-173